src/instance/minecraft/chat-manager.ts

```diff
@@ -101,8 +103,12 @@ export default class ChatManager extends EventHandler<MinecraftInstance> {
         assert(message) // old packet means message exist
         resultMessage = prismChat.fromNotch(message)
       }
-
-      this.onMessage(resultMessage.toString())
+      this.onMessage(
+        this.proceedFiltering(
+          (data as { senderName?: string }).senderName ?? 'unknown sender',
```

Comment:
`senderName` field does not exist. I assume you saw the code above it directly and thought all messages have `senderName`.
`senderName` along many other fields are only available when working on minecraft version 1.19+. It is the update that added signed messages to ensure player messages can't be faked from server side. 

You can't assume the field always exists, especially since the message can be a server message and not a player message. And that is let alone assuming the bot will always run on 1.19+.